### DISCUSSION TITLE
Unable to add alerting on rest endpoint pulled via infinity json plugin

### DISCUSSION THREAD

**Discussion by kunal-parsewar** [grafana/grafana-infinity-datasource#899] [Q&A]
Hi ,

I have configured the json infinity plugin to pull the  service health endpoint which returns json data . I was able to create the dashboard with that but not the issues are there while creating alert for the same data source health endpoint json data.

whenever I am trying to create the alert with infinity data source it is giving error as below

"[sse.readDataError] [A] got error: input data must be a wide series but got type not (input refid)"

did anyone got this error? and I am not able to figure out how to create the alert for data returned by the infinity plugin.

Grafana Version : "10.4.1"
Infinity Version : 2.7.1

@yesoreyeram

**Comment by yesoreyeram** [comment]
For alerts to work, the query must return data in numeric/timeseries format. Read more about the formats [here](https://grafana.com/developers/dataplane).

You can use Rootselector(JSONatata) or summmarize option to convert the original data into alert compatible data

**Comment by kunal-parsewar** [comment]
thanks for your reply. Now the problem is my rest endpoint send data in json string like service health ( UP /DOWN ) . I wanted to make alert based on this but not getting any idea as it is non-numeric. Do you have any idea how I can set alert for that ? any references will be helpful.

**Reply by yesoreyeram** [reply]
you can create them in multiple ways. one way is using computed column and the example is given below

### Query

<img width="1713" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/4c89046b-2577-4c6b-a727-38be1e083196">

### Alerting

<img width="1324" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/6dbbd1b9-b0b8-4cc1-b247-ba8155c95eca">

**Reply by kunal-parsewar** [reply]
cool ,thanks for the above input will try this.

**Reply by kunal-parsewar** [reply]
Thanks , above did work and was able to set the alert.
